internal/netsync/manager.go

```diff
@@ -693,8 +691,13 @@ BlockHashes:
 	for pp, requestQueue := range requestQueues {
 		var numRequested int32
 		gdmsg := wire.NewMsgGetData()
-		for _, iv := range requestQueue {
-			gdmsg.AddInvVect(iv)
+		for i := range requestQueue {
+			// Note the copy is intentional here to avoid keeping a reference
+			// into the request queue map from the message queue since that
+			// reference could potentially prevent the map from being garbage
+			// collected for an extended period of time.
+			ivCopy := requestQueue[i]
```

Comment:
Inline with your comment here: https://github.com/decred/dcrd/pull/3165/commits/c1130a9ec0695cf7adf586eeb5f2bd05b4a49cad#r1274223919, this should be `&requestQueue[i]`?

Reply:
The one is intentionally a copy per the comment.  This particular instance adds whatever is passed in to a send queue which could potentially take a while and if a pointer directly into the request queue map were passed, it would prevent GC of the entire map.

Reply:
Oh, yh. Sorry, I didn't read the comment :)

Reply:
I should note that, as has been known for a while, in hindsight, much of the `wire` API is not very efficient as it pertains to memory allocations because it was among one of the first packages written while we were still learning the intricacies of Go's GC semantics.  This is one such case where the queuing methods really shouldn't be taking pointers to avoid these types of potential footguns.

Unfortunately, changing the APIs would be breaking changes that would necessitate a new major module version and `wire` sits at the base of a ton of packages, so a new major module version is something that would cause a massive amount of churn in probably over a hundred consumers.

Given the overall minor nature of saving a few allocations versus the major churn changing it would cause, we've opted not to make any changes until/unless something that really requires a new major module versions comes along that such changes could piggyback on.

Reply:
map allocations are not addressable; i don't think there is any leak that would prevent the entire map from being garbage collected.